src/implementation.js

```diff
@@ -0,0 +1,129 @@
+function _foldToObject(memo, [k, v]) {
+  memo[k] = v;
+  return memo;
+}
+
+function _objectMap(obj, fn) {
+  return _entries(obj).map(fn).reduce(_foldToObject, {});
+}
+
+function _entries(obj) {
+  let stringKeys = Object.getOwnPropertyNames(obj);
+  let symKeys = Object.getOwnPropertySymbols(obj);
+  return [...stringKeys, ...symKeys].map(p => [p, obj[p]]);
```

Comment:
In that case, `Array.from(stringKeys, p => [p, obj[p]]).concat(Array.from(symKeys, p => [p, obj[p]]))` would be slightly more efficient

Reply:
It's a reference implementation.

Reply:
fair enough :-) altho i'd say if it's published on npm, then it's not a reference implementation - it's a production-usable package.

Reply:
Hoooo boy that is a low bar. 99.9% of the code on npm is unusable garbage that you should never *ever* put into production.